django/db/models/base.py

```diff
@@ -461,7 +463,13 @@ def __init__(self, db=None):
 class Model(six.with_metaclass(ModelBase)):
 
     def __init__(self, *args, **kwargs):
-        signals.pre_init.send(sender=self.__class__, args=args, kwargs=kwargs)
+        # Alias some things as function locals to avoid repeat global lookups
```

Comment:
I'm curious if there's a guideline about when to use these techniques. I'm particularly surpised at things like `_setattr = setattr`, though I haven't done much micro-optimization.